jcl/src/java.base/share/classes/java/lang/invoke/DirectHandle.java

```diff
@@ -117,7 +117,7 @@ private void addHandleToClassCache() {
 	}
 
 	final void nullCheckIfRequired(Object receiver) throws NullPointerException {
-		if ((receiver == null) && !Modifier.isStatic(final_modifiers)) {
+		if (!isStatic && (receiver == null)) {
```

Comment:
Originally this code checked the `receiver ==null` first as it was a cheaper check than `Modifier.isStatic()`.  @andrewcraik Any opinion on whether it's worth keeping the explicit nullcheck or not?

Reply:
I don't think you need it. @liqunl will removing it mess up any patterns? If anything I would have expected the removal to improve things...

Reply:
Thanks.  I've removed it and merged the DirectH & RBH null check methods.

Reply:
Sorry, I was working on a defect. The removal looks good to me.